tests/unit/Utils/DPOPTest.php

```diff
@@ -0,0 +1,155 @@
+<?php
+
+namespace Pdsinterop\Solid\Auth\Utils;
+
+use PHPUnit\Framework\TestCase;
+use Lcobucci\JWT\Validation\RequiredConstraintsViolated;
+
+/**
+ * @covers DPop
+ */
+class DPOPTest extends TestCase
+{
+
+	private $dpop;
+	private $url;
+	private $serverRequest;
+
+	protected function sign($dpop, $privateKey=null)
+	{
+		$keyPath    = dirname(__DIR__) . '/../fixtures/keys';
+		if (!$privateKey) {
+	        $privateKey = file_get_contents($keyPath . '/private.key');
+	    }
+
+		$signature = '';
+        $success = \openssl_sign(
+            Base64Url::encode(json_encode($dpop['header'])).'.'.
+            Base64Url::encode(json_encode($dpop['payload'])), 
+            $signature, 
+            $privateKey, 
+            OPENSSL_ALGO_SHA256
+        );
+
+        if (!$success) {
+        	throw new \Exception('unable to sign dpop');
+        }
+        $token = Base64Url::encode(json_encode($dpop['header'])).'.'.
+            	 Base64Url::encode(json_encode($dpop['payload'])).'.'.
+            	 Base64Url::encode($signature);
+
+        return array_merge($dpop, [
+        	'signature' => $signature,
+        	'token' => $token
+        ]);
+	}
+
+	protected function setUp(): void
+	{
+		$keyPath    = dirname(__DIR__) . '/../fixtures/keys';
```

Comment:
This looks weird to me... 

I'd say, either use this:
```php
$keyPath = dirname(__DIR__, 2) . '/fixtures/keys';
```
or this:
```
$keyPath = __DIR__ . '/../../fixtures/keys';
```

I wouldn't mix `dirname` and `../`

Reply:
done